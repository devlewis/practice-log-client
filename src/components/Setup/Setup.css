.Setup_container {
  display: flex;
  flex-direction: column;
  max-width: 75%;
  width: fit-content;
  margin: auto;
}

.Setup_buttons {
  display: flex;
  justify-content: flex-end;
  margin-top: 20px;
  width: auto;
}

.Setup_logout {
  margin-left: 10px;
}
