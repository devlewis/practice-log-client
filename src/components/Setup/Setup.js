import React, { PureComponent } from "react";
import "../Setup/Setup.css";
import Context from "../../Context";
import TokenService from "../../services/token-service";
import IdleService from "../../services/idle-service";

class Setup extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      num_of_days: null,
      hours: null,
    };
  }

  static contextType = Context;

  handleSubmit = (e) => {
    e.preventDefault();
    this.context.onHandleSubmit(
      this.state.num_of_days,
      this.state.hours,
      this.props.history
    );
  };

  handleLogout = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
    this.props.history.push("/");
  };

  render() {
    return (
      <div className="Setup_container">
        <h2>Setup</h2>
        <form className="Setup_form" onSubmit={this.handleSubmit}>
          <div className="Select_box">
            <label htmlFor="#ofDaysChoice"># of Practice Days</label>
            <select
              onChange={(e) =>
                this.setState({ num_of_days: parseFloat(e.target.value) })
              }
              name="#ofDaysChoice"
              id="#ofDaysChoice"
              required
            >
              <option value="">Select Length of Goal</option>
              <option value="7">7 Days</option>
              <option value="30">30 Days</option>
              <option value="100">100 Days</option>
            </select>
          </div>
          <div>
            <label htmlFor="hours"> Hours Per Day </label>
            <input
              onChange={(e) =>
                this.setState({
                  hours: parseFloat(e.target.value),
                })
              }
              type="text"
              name="hours"
              id="hours"
              placeholder="# of hrs"
              autoComplete="off"
              required
            />
          </div>
          <div className="Setup_buttons">
            <button type="submit">Start my goal today!</button>
          </div>
        </form>
      </div>
    );
  }
}

export default Setup;
