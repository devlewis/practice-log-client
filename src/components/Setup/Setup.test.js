import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Setup from "./Setup";

describe(`Setup component`, () => {
  const props = {
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop",
  };

  it("renders form.Setup by default", () => {
    const wrapper = shallow(<Setup />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Setup given props", () => {
    const wrapper = shallow(<Setup {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
