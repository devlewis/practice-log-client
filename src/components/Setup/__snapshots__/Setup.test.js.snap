// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setup component renders form.Setup by default 1`] = `
<div
  className="Setup_container"
>
  <h2>
    Setup
  </h2>
  <form
    className="Setup_form"
    onSubmit={[Function]}
  >
    <div
      className="Select_box"
    >
      <label
        htmlFor="#ofDaysChoice"
      >
        # of Practice Days
      </label>
      <select
        id="#ofDaysChoice"
        name="#ofDaysChoice"
        onChange={[Function]}
        required={true}
      >
        <option
          value=""
        >
          Select Length of Goal
        </option>
        <option
          value="7"
        >
          7 Days
        </option>
        <option
          value="30"
        >
          30 Days
        </option>
        <option
          value="100"
        >
          100 Days
        </option>
      </select>
    </div>
    <div>
      <label
        htmlFor="hours"
      >
         Hours Per Day 
      </label>
      <input
        autoComplete="off"
        id="hours"
        name="hours"
        onChange={[Function]}
        placeholder="# of hrs"
        required={true}
        type="text"
      />
    </div>
    <div
      className="Setup_buttons"
    >
      <button
        disabled={false}
        type="submit"
      >
        Start my goal today!
      </button>
    </div>
  </form>
</div>
`;

exports[`Setup component renders the Setup given props 1`] = `
<div
  className="Setup_container"
>
  <h2>
    Setup
  </h2>
  <form
    className="Setup_form"
    onSubmit={[Function]}
  >
    <div
      className="Select_box"
    >
      <label
        htmlFor="#ofDaysChoice"
      >
        # of Practice Days
      </label>
      <select
        id="#ofDaysChoice"
        name="#ofDaysChoice"
        onChange={[Function]}
        required={true}
      >
        <option
          value=""
        >
          Select Length of Goal
        </option>
        <option
          value="7"
        >
          7 Days
        </option>
        <option
          value="30"
        >
          30 Days
        </option>
        <option
          value="100"
        >
          100 Days
        </option>
      </select>
    </div>
    <div>
      <label
        htmlFor="hours"
      >
         Hours Per Day 
      </label>
      <input
        autoComplete="off"
        id="hours"
        name="hours"
        onChange={[Function]}
        placeholder="# of hrs"
        required={true}
        type="text"
      />
    </div>
    <div
      className="Setup_buttons"
    >
      <button
        disabled={false}
        type="submit"
      >
        Start my goal today!
      </button>
    </div>
  </form>
</div>
`;
